client-app/src/pages/account/components/invoices-filter/invoices-filter.ts

```diff
@@ -0,0 +1,110 @@
+import Vue from "vue";
+import Component from "vue-class-component";
+import { LocaleMessages, TranslateResult } from "vue-i18n";
+import { Prop } from "vue-property-decorator";
+import { faAngleUp, faAngleDown } from "@fortawesome/free-solid-svg-icons";
+import i18n from "@i18n";
+import { InvoicesListFilters } from "@account/store/modules/invoices-list/types";
+import { locale } from "@common/constants";
+
+@Component
+export default class InvoicesFilter extends Vue {
+  @Prop()
+  invoicesFilter!: InvoicesListFilters;
+
+  @Prop()
+  availableInvoicesStatuses!: string[];
+
+  startDate: Date | undefined = undefined;
+
+  endDate: Date | undefined = undefined;
+
+  isDateValid: boolean | null = null;
+
+  allStatusesSelected = false;
+
+  selectedStatuses: string[] = [];
+
+  datepickerLabels: LocaleMessages | {} = {};
+
+  locale = locale;
+
+  faAngleUp = faAngleUp;
+
+  faAngleDown = faAngleDown;
+
+  showStatusDropdown = false;
+
+  emitChanges(updatedFilters: InvoicesListFilters) {
+    this.$emit("filtersChanged", updatedFilters);
+  }
+
+  mounted() {
+    this.getDatepickerLocalization();
+  }
+
+  dateChanged() {
+    if (this.startDate && this.endDate) {
+      this.isDateValid = this.startDate <= this.endDate;
+      if (this.isDateValid) {
+        const updatedFilters = { ...this.invoicesFilter, startDate: this.startDate, endDate: this.prepareEndDate() };
+        this.emitChanges(updatedFilters);
+      }
+    } else {
+      this.endDate == null && this.startDate == null ? (this.isDateValid = null) : (this.isDateValid = true);
+      const updatedFilters = {
+        ...this.invoicesFilter,
+        startDate: this.startDate,
+        endDate: this.endDate ? this.prepareEndDate() : undefined
+      };
+      this.emitChanges(updatedFilters);
+    }
+  }
+
+  changeStartDate(date: Date) {
+    this.startDate = date;
+    this.dateChanged();
+  }
+
+  changeEndDate(date: Date) {
+    this.endDate = date;
+    this.dateChanged();
+  }
+
+  getDatepickerLocalization() {
```

Comment:
If you using the verb 'Get' then your function should return something. But you set values for datepicker labels then you have to use a verb 'Set' or something like that